### PR TITLE
JS linter: Add initial support for JS linter

### PR DESCRIPTION
**Pull request by tcwanggoogle**
Only lint JS files under frontend/

### CODE REVIEW

**Comment by tcwanggoogle** [comment]
Javascript linter also added. Only applies to frontend/ directory. There are plenty of violations need fixing. Any volunteers?

**Comment by kmh287** [comment]
@dhruvi618 would you be best suited to close out the lint errors here?

**Comment by dhruvi618** [comment]
> @dhruvi618 would you be best suited to close out the lint errors here?

Sure, I can do it.

**Comment by tcwanggoogle** [comment]
> > @dhruvi618 would you be best suited to close out the lint errors here?
> 
> Sure, I can do it.

Thanks! Shared a doc with you with instructions on how to run this linter locally (so you don't have to create a PR to see if you fix the problems or not)

**Comment by dhruvi618** [comment]
Finished fixing all JS lint errors (I merged my original branch `fix-js-lint-errors` into this branch to show/verify that all lint errors were fixed). Need everyone's LGTM to eventually merge into master.

**Comment by tcwanggoogle** [comment]
Awesome! Thanks for the fix. It looks like I am the author, so I can't approve. Will need others' approval

File: .github/workflows/eslint.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install npm packages
+        run: npm install eslint@latest --save-dev
+      - name: Install eslint React Plugin
+        run: npm install eslint-plugin-react@latest --save-dev
+      - name: Install Google style plugin
+        run: npm install eslint-config-google@latest --save-dev
+      - name: Lint with ESLint
+        run: ./node_modules/.bin/eslint frontend/**/*.js
```
Comment (TcgVanguardTroll) [inline .github/workflows/eslint.yaml:18]: This might be a nit but can you explain what this file does ?

Comment (dhruvi618) [reply .github/workflows/eslint.yaml:18]: Sure! So in order to successfully run the linter, there are some preconditions which need to be satisfied (i.e plugins, packages, and certain configurations that need to be installed). This file makes sure that the correct plugins and packages are installed for the .eslintrc.json file to correctly run and actually runs the necessary commands to lint. Side note: If you look at the details in the JS lint check on this PR, you can see these steps in the log.

Comment (tcwanggoogle) [reply .github/workflows/eslint.yaml:18]: This yaml file, like other linter/formatter, is for us to configure a Ubuntu machine to tell Github to run a "workflow" during the "[pull_request]". All such files should go to .github/workflows(https://docs.github.com/en/actions/configuring-and-managing-workflows/configuring-a-workflow). For this specific file, we want to configure a workflow to run eslint, which is a JavaScript linter, during pull request. Like Dhruvi said, the steps before the eslint is to make sure the dependencies are installed on that machine Github provides us.

**Comment by tcwanggoogle** [comment]
Will merge this PR with 3 approvals.

**Pull request merged**